pandas-stubs/core/dtypes/base.pyi

```diff
@@ -1,23 +1,22 @@
-from typing import Literal
+from typing import (
+    Any,
+    Literal,
+)
 
 from pandas.core.arrays import ExtensionArray
 
-from pandas._libs import NaTType
-from pandas._libs.missing import NAType
 from pandas._typing import type_t
 
 class ExtensionDtype:
-    @property
-    def na_value(self) -> NAType | NaTType: ...
-    @property
-    def type(self) -> type_t: ...
+    type: type_t
+    na_value: Any
```

Comment:
`na_value` seems to be a property (but the return value can be Any for some classes)

Reply:
OK, had to fix the base class to make them abstract properties

Reply:
I wouldn't be surpsrised if they are used inconsistently within pandas :( in some places as properties, in some as normal attributes and in some as class variables. I just grepped through pandas/core/dtypes, I didn't find a single line that assigns a value to a class variable `na_value` or `type`. It seems consistently to be a property.

If they are class variables, can you add a quick `isinstance(ClassName.na_value, ...)` test.

If the pandas tests set it for simplicity/laziness to a class varaible (which is fine with me), we can just add an ignore to them.

Reply:
> I wouldn't be surpsrised if they are used inconsistently within pandas :( in some places as properties, in some as normal attributes and in some as class variables. I just grepped through pandas/core/dtypes, I didn't find a single line that assigns a value to a class variable `na_value` or `type`. It seems consistently to be a property.

Take a look at `pandas/core/arrays/integer.py`.  You'll see assignments of `type = ` for the various `Integer` dtypes.

I believe it's declared as a property in `pandas/core/dtypes/base.py` so that if someone doesn't define it, it raises `AbstractMethodError`.  But it is used internally as a `ClassVar`.

For `na_value`, it is inconsistent, but based on the example for `DecimalDtype`, it was treated as a class variable there as well.

Reply:
I left `name` and `type` as `ClassVar`, but changed `na_value` to be a property, because it has a default value.

Reply:
> Take a look at `pandas/core/arrays/integer.py`. You'll see assignments of `type = ` for the various `Integer` dtypes.

I wounder why mypy/pyright didn't catch that. Maybe because a [parent class re-declares `type` as `Any`](https://github.com/pandas-dev/pandas/blob/ac70f230995aa1bbee97b5db7bfdc28e5c5ed14b/pandas/core/dtypes/dtypes.py#L82) or because of the decorators that mypy doesn't understand. Seems like something that should be fixed in pandas.

Reply:
> I wounder why mypy/pyright didn't catch that. Maybe because a [parent class re-declares `type` as `Any`](https://github.com/pandas-dev/pandas/blob/ac70f230995aa1bbee97b5db7bfdc28e5c5ed14b/pandas/core/dtypes/dtypes.py#L82) or because of the decorators that mypy doesn't understand. Seems like something that should be fixed in pandas.

I was wondering the same thing.

In reality, `type` and `name` are abstract class properties of `ExtensionDtype`, but the type checkers aren't supporting that concept.  See long discussions here:  https://github.com/microsoft/pyright/issues/2601 and https://github.com/python/mypy/issues/8993